cmd/containerd-shim-runhcs-v1/task_hcs.go

```diff
@@ -596,6 +598,54 @@ func (ht *hcsTask) DumpGuestStacks(ctx context.Context) string {
 	return ""
 }
 
+func (ht *hcsTask) Share(ctx context.Context, req *shimdiag.ShareRequest) error {
+	if ht.host == nil {
+		return errors.New("task is not isolated")
+	}
+	guestReq := guestrequest.GuestRequest{
+		ResourceType: guestrequest.ResourceTypeMappedDirectory,
+		RequestType:  requesttype.Add,
+	}
+	// If the task is WCOW then we'll be mounting the directory/file with VSMB
+	// and returning.
+	if ht.isWCOW {
```

Comment:
This function will only be called on the sandboxTask of a pod and in case of WCOW the sandboxTask is always an instance of `wcowPodSandboxTask` and so I don't think we need to handle the case of WCOW here.

Reply:
That's very true good catch. I forgot about the fake task concept.